clients/web/src/views/widgets/MetadataWidget.js

```diff
@@ -2,80 +2,421 @@
+            el: newEditRow.find('div'),
             item: this.item,
             key: '',
-            value: '',
+            value: value,
+            accessLevel: this.accessLevel,
             newDatum: true,
+            parentView: widget
+        }).render();
+    },
+
+    render: function () {
+        var metaDict = this.item.attributes.meta || {};
+        var metaKeys = Object.keys(metaDict);
+        metaKeys.sort(girder.localeSort);
+
+        // Metadata header
+        this.$el.html(girder.templates.metadataWidget({
+            item: this.item,
             accessLevel: this.accessLevel,
-            parentView: this
+            girder: girder
+        }));
+
+        // Append each metadatum
+        _.each(metaKeys, function (metaKey) {
+            this.$el.find('.g-widget-metadata-container').append(new girder.views.MetadatumWidget({
+                mode: this.getModeFromValue(metaDict[metaKey]),
+                key: metaKey,
+                value: metaDict[metaKey],
+                accessLevel: this.accessLevel,
+                girder: girder,
+                parentView: this
+            }).render().$el);
+        }, this);
+
+        this.$('.g-widget-metadata-add-button').tooltip({
+            container: this.$el,
+            placement: 'left',
+            animation: false,
+            delay: {show: 100}
         });
+
+        return this;
+    }
+});
+
+girder.views.MetadatumWidget = girder.View.extend({
+    events: {
+        'click .g-widget-metadata-edit-button': 'editMetadata'
+    },
+
+    initialize: function (settings) {
+        if (!_.has(this.parentView.modes, settings.mode)) {
+            throw 'Unsupported metadatum mode ' + settings.mode + ' detected.';
+        }
+
+        this.mode = settings.mode;
+        this.key = settings.key;
+        this.value = settings.value;
+        this.accessLevel = settings.accessLevel;
+        this.parentView = settings.parentView;
+    },
+
+    _validate: function (from, to, value) {
+        var newMode = this.parentView.modes[to];
+
+        if (_.has(newMode, 'validation') &&
+            _.has(newMode.validation, 'from') &&
+            _.has(newMode.validation.from, from)) {
+
+            var validate = newMode.validation.from[from][0];
+            var msg = newMode.validation.from[from][1];
+
+            if (!validate(value)) {
+                girder.events.trigger('g:alert', {
+                    text: msg,
+                    type: 'warning'
+                });
+                return false;
+            }
+        }
+
+        return true;
+    },
+
+    // @todo too much duplication with editMetadata
+    toggleEditor: function (event, newEditorMode, existingEditor, overrides) {
+        var fromEditorMode = (existingEditor instanceof girder.views.JsonMetadatumEditWidget) ? 'json' : 'simple';
+        if (!this._validate(fromEditorMode, newEditorMode, (overrides || {}).value || existingEditor.$el.attr('g-value'))) {
```

Comment:
Done, I noticed there is no JS max line length in our jshint - is that something we should change?